### PR TITLE
add HCAL scale monitoring, create PFCandidate Analyzer

### PR DESCRIPTION
**Pull request by webermat**
create PFCandidateAnalyzer, move existing PFCandidate Monitoring histograms from METAnalyzer into newly created module. Add HCAL scale monitoring based on isolated charged hadron candidates comparing HCAL Cluster Energies and Track momentum. Reduce number of bins in phi for PFCandidate eta-phi maps.

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by webermat** [comment]
the test results look fine for some workflows, but for other workflows (combining MiniAOD and RECO workflows in one go) some histograms fail to be filled. I hope to find out why soon.

**Comment by webermat** [comment]
the problem should be fixed now: a parameter was not filled correctly. Since DQM runs now MiniAOD and RECO in parallel top level plots were filled twice, that has been corrected too.

**Comment by webermat** [comment]
please test

**Comment by deguio** [comment]
please test

**Comment by webermat** [comment]
+1
comparisons look fine now

**Comment by deguio** [comment]
+1

**Comment by davidlange6** [comment]
@webermat - apologies - this one slipped off of my list and now does not merge - could you rebase it?

**Comment by webermat** [comment]
the rebased version is in PR  #14422

**Comment by webermat** [comment]
as the rebased version could not be pushed back easily I created a new PR after the rebase of this PR in #14422

File: DQMOffline/JetMET/src/PFCandidateAnalyzer.cc [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+      for(unsigned int i=0;i<FilterhltConfig_.size();i++){
+	std::string search=FilterhltConfig_.triggerName(i).substr(5);//actual label of filter, the first 5 items are Flag_, so stripped off
+	std::string search2=HBHENoiseStringMiniAOD;//all filters end with DQM, which is not in the flag --> ONLY not for HBHEFilters
+	std::size_t found=search2.find(search);
+	if(found!=std::string::npos){
+	  miniaodfilterindex=i;
+	}
+      }
+    }else{
+      std::cout<<"nothing found with both RECO and reRECO label"<<std::endl;
```
Comment (dmitrijus) [inline DQMOffline/JetMET/src/PFCandidateAnalyzer.cc]: either prefix it with "if (verbose_)" or, even better, move to LogWarning

**Review by dmitrijus — APPROVED** [review]

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
please test

**Comment by dmitrijus** [comment]
hold

**Comment by webermat** [comment]
there was one more issue in previous tests which should be corrected now. This issue was present in a more complete build and had two instances of the plugin 'PFCandidateAnalyzer. This should be corrected now by renaming the class. 
Another possible issue had been corrected 4 days ago as well, which was not taken care off previously: https://github.com/cms-sw/cmssw/commit/55ac8f20973cafe770178ca7d11b59f558eaa5ef

**Comment by dmitrijus** [comment]
unhold
please resolve the conflict, sorry it takes so long :(

**Comment by webermat** [comment]
the conflict should have been resolved. I checked previously that the 3 workflows from above (10424.0 etc) run successfully now, ttbar workflow 25.0 shows the desired result as well.

**Comment by dmitrijus** [comment]
+1

**Comment by webermat** [comment]
comparisons look fine (based e.g. on TTbar workflows 25 and 25202)

File: DQMOffline/JetMET/src/DQMPFCandidateAnalyzer.cc [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+  }
+  if(bDCSFilter && hbhenoifilterdecision && bPrimaryVertex){
+    if(isMiniAOD_){
+      edm::Handle<std::vector<pat::PackedCandidate> > packedParticleFlow;
+      iEvent.getByToken(pflowPackedToken_, packedParticleFlow);
+      //11, 13, 22 for el/mu/gamma, 211 chargedHadron, 130 neutralHadrons, 1 and 2 hadHF and EGammaHF
+      for (unsigned int i = 0; i < packedParticleFlow->size(); i++) {
+	const pat::PackedCandidate& c = packedParticleFlow->at(i);
+	for (unsigned int j=0; j<typePFCand_.size(); j++) {
+	  if (fabs(c.pdgId())==typePFCand_[j]) {
```
Comment (davidlange6) [inline DQMOffline/JetMET/src/DQMPFCandidateAnalyzer.cc]: @webermat - this should be abs (or just std::abs)

**Comment by davidlange6** [comment]
please test
modernizing test results - code review is ok aside from a one line fix needed

**Comment by davidlange6** [comment]
hi @webermat - there was a small change needed for this pr - but now it needs a rebase. Please update your branch and we'll integrated into 90x via #16650.

**Pull request closed without merging**